src/libraries/System.Linq.Expressions/src/System/Dynamic/ExpandoObject.cs

```diff
@@ -317,6 +317,14 @@ private bool ExpandoContainsKey(string key)
             return _data.Class.GetValueIndexCaseSensitive(key) >= 0;
         }
 
+        // This implementation is shared for 'IDictionary<,>.ContainsKey' and 'IReadOnlyDictionary<,>.ContainsKey'.
+        private bool ExpandoContainsKeyNoLock(string key)
```

Comment:
It is confusing to call this `ExpandoContainsKeyNoLock` since it does not do the same thing as `ExpandoContainsKey`

Reply:
I've renamed it to match the exact semantics of the helper 🙂